Sources/MediaViewController.swift

```diff
@@ -112,20 +128,21 @@ public class VLCMediaViewController: UICollectionViewController, UISearchResults
                 backgroundview.clipsToBounds = true
             }
         }
-        if #available(iOS 11.0, *) {
-            navigationItem.searchController = searchController
-        } else {
-            navigationItem.titleView = searchController?.searchBar
-            searchController?.hidesNavigationBarDuringPresentation = false
-        }
+        showSearchController(true)
```

Comment:
Actually, don't have this state with true. It should be determined by the method itself and call it showSearchControllerIfNeeded. In there you will check the collectionview.numberOfSections